packages/gitlab-backend/src/processor/projectIdProcessor.ts

```diff
@@ -0,0 +1,66 @@
+import { CatalogProcessor } from '@backstage/plugin-catalog-node';
+import { Entity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { CacheService } from '@backstage/backend-plugin-api';
+import {
+    GITLAB_INSTANCE,
+    GITLAB_PROJECT_ID,
+    GITLAB_PROJECT_SLUG,
+} from '../annotations';
+
+/** @public */
+export class GitlabProjectIdProcessor implements CatalogProcessor {
+    constructor(
+        private readonly config: Config,
+        private readonly cache: CacheService
+    ) {}
+
+    getProcessorName(): string {
+        return 'GitlabProjectIdProcessor';
+    }
+
+    async postProcessEntity(entity: Entity): Promise<Entity> {
+        const annotations = entity.metadata?.annotations;
+        const slug = annotations?.[GITLAB_PROJECT_SLUG];
+        const host = annotations?.[GITLAB_INSTANCE];
+
+        if (!slug || !host || annotations?.[GITLAB_PROJECT_ID] !== undefined) {
```

Comment:
This check is not ok, if the project slug or host change, even if `GITLAB_PROJECT_ID` is defined, we have to reevaluate it, so you cannot use: `annotations?.[GITLAB_PROJECT_ID] !== undefined`. That's the reason for the cache.